test/mocks/MockReenterHook.sol

```diff
@@ -21,8 +22,8 @@ contract MockReenterHook is BaseTestHooks {
         }
         (bytes4 selector, address owner, uint256 tokenId) = abi.decode(functionSelector, (bytes4, address, uint256));
 
-        if (selector == posm.transferFrom.selector) {
-            posm.transferFrom(owner, address(this), tokenId);
+        if (selector == IERC721(address(posm)).transferFrom.selector) {
```

Comment:
why doesnt IPositionManager just inherit from IERC721?

Reply:
I can't really use an interface that defines the events because the solmate ERC721 implementation already includes the events and doesn't have an interface. So if `IPositionManager` inherits from `IERC721` I get an error that the event with the same name and parameter types was defined twice. The only way around that would be to create a custom ERC721 interface without the events, so I opted for casting in the tests instead